### DISCUSSION TITLE
Bluetooth device getting automatically disconnected upon connection

### DISCUSSION THREAD

**Discussion by NikhileshWaniTech** [hbldh/bleak#881] [Q&A]
Hello! I am using an nrf52 device to connect with a windows 11 machine using the uart_service.py example code available in this repo. However, after connecting the device it is getting automatically disconnected. 
As the same device works properly with a web application using Bluetooth Web APIs, the possibility of hardware issues can be safely eliminated. 
The way it works with the web is that upon connecting the device, the device sends its ID through uart and then the web application sends some commands to the device. I am trying to implement the same functionality with the python app.
Kindly suggests a possible solution. 

Thanks!

**Comment by dlech** [comment]
Did you modify the example program? Are there any error messages printed when it disconnects? Have you tried troubleshooting steps from docs like enabling logging in Python or capturing Bluetooth packets with Wireshark?
